packages/lib/gpcircuits/src/util.ts

```diff
@@ -11,3 +12,74 @@ export function artifactPaths(
     vkeyPath: path.join(root, cd.name + "-vkey.json")
   };
 }
+
+/**
+ * Helpers
+ */
+
+// Procedure for splitting an array `arr` into chunks of size
+// `n` in order. If `arr.length` is not a multiple of `n`,
+// then the last chunk will be of length `arr.length % n`.
+export function toChunks<A>(arr: A[], n: number): A[][] {
+  return arr.reduce(
+    (chunks: A[][], a: A) => {
+      const lastChunkIndex = chunks.length - 1;
+      const lastChunk = chunks[lastChunkIndex];
+      return lastChunk.length < n // If the last chunk is incomplete
+        ? chunks.slice(0, lastChunkIndex).concat([lastChunk.concat([a])]) // Append the current value to it.
+        : chunks.concat([[a]]); // Else start a new chunk with the current value in it.
+    },
+    [[]]
+  );
+}
+
+// Procedure for applying a Promise-valued function to
+// array elements sequentially. Necessary to avoid OOM
+// for particularly heavy computations.
+export function seqPromise<A, B>(
```

Comment:
What's the core purpose here?  I think this is complicated code for something very simple, so I wonder if I'm missing something.

If this were written with async/await I think this would be equivalent to:
```
  return arr.map((a) => await f(a));
```
which is simple enough to not even need a helper function.  Am I right about that equivalency, or is there a subtle difference I'm missing?